features/storage/system_storage/SystemStorage.cpp

```diff
@@ -0,0 +1,62 @@
+#include "SystemStorage.h"
+
+namespace mbed {
+namespace SystemStorage{
+
+MBED_WEAK BlockDevice* get_default_blockdevice()
+{
+#if COMPONENT_SPIF
+
+    static SPIFBlockDevice default_bd(
+        MBED_CONF_SPIF_DRIVER_SPI_MOSI, 
+        MBED_CONF_SPIF_DRIVER_SPI_MISO,
+        MBED_CONF_SPIF_DRIVER_SPI_CLK, 
+        MBED_CONF_SPIF_DRIVER_SPI_CS, 
+        MBED_CONF_SPIF_DRIVER_SPI_FREQ
+        );
+
+    return &default_bd;
+
+#elif COMPONENT_SD
+
+    static SDBlockDevice default_bd(
+        MBED_CONF_SD_SPI_MOSI, 
+        MBED_CONF_SD_SPI_MISO,
+        MBED_CONF_SD_SPI_CLK, 
+        MBED_CONF_SD_SPI_CS
+        );
+
+    return &default_bd;
+
+#else
+
+    return NULL;
+
+#endif
+
+}
+
+MBED_WEAK FileSystem * get_default_filesystem()
```

Comment:
This looks great to me 👍

Assuming there's no issues with weak functions in this context. @theotherjimmy are you able to double check?